tests/functionnal/SLM.php

```diff
@@ -399,4 +622,158 @@ public function testInternalTtrComputation() {
+      $this->variable($ticket->fields['internal_time_to_resolve'])->isEqualTo($after1month);
+
+      $this->integer((int)$ticket->fields['olas_id_tto'])->isEqualTo($ola_tto_id);
+      $this->variable($ticket->fields['internal_time_to_own'])->isEqualTo($after1month);
+
+      $this->integer((int)$ticket->fields['slas_id_ttr'])->isEqualTo($sla_ttr_id);
+      $this->variable($ticket->fields['time_to_resolve'])->isEqualTo($after1month);
+
+      $this->integer((int)$ticket->fields['slas_id_tto'])->isEqualTo($sla_tto_id);
+      $this->variable($ticket->fields['time_to_own'])->isEqualTo($after1month);
+
+      $this->integer(strtotime($ticket->fields['ola_ttr_begin_date']))
+         ->isGreaterThanOrEqualTo($createtime)
+         ->isLessThanOrEqualTo($createtime);
+
+      // Create ticket to test computation based on OLA / SLA on update
+      $ticket = new \Ticket();
+      $ticket_id = $ticket->add(
+         [
+            'name'    => 'Test Ticket',
+            'content' => 'Ticket for TTR OLA test on update',
+         ]
+      );
+      $this->integer($ticket_id)->isGreaterThan(0);
+
+      $this->boolean($ticket->getFromDB($ticket_id))->isTrue();
+      $this->integer((int)$ticket->fields['olas_id_ttr'])->isEqualTo(0);
+      $this->variable($ticket->fields['ola_ttr_begin_date'])->isEqualTo(null);
+      $this->variable($ticket->fields['internal_time_to_resolve'])->isEqualTo(null);
+
+      //Wait...
+      sleep(1);
+
+      // Assign TTR/TTO OLA/SLA
+      $update_time_1 = time();
+      $this->boolean($ticket->update(
+         ['id' => $ticket_id,
+          'olas_id_ttr' => $ola_ttr_id,
+          'olas_id_tto' => $ola_tto_id,
+          'slas_id_ttr' => $sla_ttr_id,
+          'slas_id_tto' => $sla_tto_id,
+         ]))->isTrue();
+      $update_time_2 = time();
+
+      $after1month = date('Y-m-d H:i', strtotime($ticket->fields['date'].'+1 month'));
+      $this->boolean($ticket->getFromDB($ticket_id))->isTrue();
+
+      //check computed data from SLA / OLA
+      $this->integer((int)$ticket->fields['olas_id_ttr'])->isEqualTo($ola_ttr_id);
+      $this->variable(date('Y-m-d H:i', strtotime($ticket->fields['internal_time_to_resolve'])))->isEqualTo($after1month);
+
+      $this->integer((int)$ticket->fields['olas_id_tto'])->isEqualTo($ola_tto_id);
+      $this->variable(date('Y-m-d H:i', strtotime($ticket->fields['internal_time_to_own'])))->isEqualTo($after1month);
+
+      $this->integer((int)$ticket->fields['slas_id_ttr'])->isEqualTo($sla_ttr_id);
+      $this->variable(date('Y-m-d H:i', strtotime($ticket->fields['time_to_resolve'])))->isEqualTo($after1month);
+
+      $this->integer((int)$ticket->fields['slas_id_tto'])->isEqualTo($sla_tto_id);
+      $this->variable(date('Y-m-d H:i', strtotime($ticket->fields['time_to_own'])))->isEqualTo($after1month);
+
+      $this->integer(strtotime($ticket->fields['ola_ttr_begin_date']))
+         ->isGreaterThanOrEqualTo($update_time_1)
+         ->isLessThanOrEqualTo($update_time_2);
```

Comment:
May be you can check that it is equal to `date_mod` field of the ticket.

Reply:
Fixed